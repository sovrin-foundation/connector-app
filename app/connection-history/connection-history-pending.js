// @flow
import React, { PureComponent } from 'react'
import { connect } from 'react-redux'
import { Container } from '../components'
import type { ReactNavigation } from '../common/type-common'
import type { Store } from '../store/type-store'
import { ClaimRequestStatusModal } from '../claim-offer/claim-request-modal'
import { CLAIM_REQUEST_STATUS } from '../claim-offer/type-claim-offer'

type ConnectionHistoryPendingProps = {} & ReactNavigation

export class ConnectionHistoryPending extends PureComponent<
  ConnectionHistoryPendingProps,
  void
> {
  close = () => {
    this.props.navigation.goBack()
  }

  render() {
    const { payload } = this.props.navigation.state
      ? this.props.navigation.state.params
      : {}

    return (
      <Container fifth>
        <ClaimRequestStatusModal
          fromConnectionHistory
          claimRequestStatus={CLAIM_REQUEST_STATUS.SEND_CLAIM_REQUEST_SUCCESS}
          payload={payload}
          onContinue={this.close}
          senderLogoUrl={payload.senderLogoUrl}
          isPending={true}
          message1="As soon as"
          message3="signs and issues the credential"
          message5="to you, it will appear here."
        />
      </Container>
    )
  }
}
