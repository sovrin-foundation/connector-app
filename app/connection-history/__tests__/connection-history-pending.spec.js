// @flow
import React from 'react'
import 'react-native'
import renderer from 'react-test-renderer'
import { Provider } from 'react-redux'
import { ConnectionHistoryPending } from '../connection-history-pending'
import {
  getStore,
  senderLogoUrl,
  getNavigation,
} from '../../../__mocks__/static-data'

describe('<ConnectionHistoryPendings />', () => {
  const store = getStore()
  let getProps = () => ({
    navigation: getNavigation({
      payload: {
        issuer: 'issuer',
        data: { name: 'requester' },
        senderLogoUrl,
      },
    }),
  })

  it('should ConnectionHistoryPending render properly', () => {
    let props = getProps()
    const component = renderer.create(
      <Provider store={store}>
        <ConnectionHistoryPending {...props} />
      </Provider>
    )
    expect(component).toMatchSnapshot()
  })
  it('should redirect previous screen is close is called', () => {
    let props = getProps()
    const component = renderer.create(
      <Provider store={store}>
        <ConnectionHistoryPending {...props} />
      </Provider>
    )
    const componentInstance = component.root.findByType(
      ConnectionHistoryPending
    ).instance
    componentInstance.close()
    expect(props.navigation.goBack).toHaveBeenCalled()
  })
})
