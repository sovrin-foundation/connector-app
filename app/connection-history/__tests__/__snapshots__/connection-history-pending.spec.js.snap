// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ConnectionHistoryPendings /> should ConnectionHistoryPending render properly 1`] = `
<View
  fifth={true}
  style={
    Object {
      "backgroundColor": "#ffffff",
      "flex": 1,
    }
  }
>
  <View
    center={true}
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
      }
    }
  >
    <View
      center={true}
      shadow={true}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#ffffff",
          "elevation": 10,
          "justifyContent": "center",
          "marginHorizontal": 30,
          "shadowColor": "rgba(0, 0, 0, 0.2)",
          "shadowOffset": Object {
            "height": 10,
            "width": 0,
          },
          "shadowOpacity": 1,
          "shadowRadius": 30,
        }
      }
    >
      <View
        center={true}
        style={
          Object {
            "alignItems": "center",
            "borderBottomColor": "#3f3f3f",
            "borderBottomWidth": 0.2,
            "justifyContent": "center",
            "padding": 20,
            "paddingVertical": 8,
          }
        }
        verticalSpace={true}
      >
        <View
          accessibilityLabel="claim-request-text-avatars-container"
          accessible={true}
          row={true}
          spaceBetween={true}
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "justifyContent": "space-between",
              "marginHorizontal": 10,
              "marginVertical": 10,
            }
          }
          testID="claim-request-text-avatars-container"
          vCenter={true}
        >
          <View
            accessibilityComponentType={undefined}
            accessibilityLabel={undefined}
            accessibilityTraits={undefined}
            accessible={true}
            hitSlop={undefined}
            nativeID={undefined}
            onLayout={undefined}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            pointerEvents="box-only"
            style={undefined}
            testID={undefined}
          >
            <View
              accessibilityLabel="avatars-pair-avatars-invitee"
              accessible={true}
              imageShadow={null}
              style={
                Object {
                  "borderRadius": 38,
                }
              }
              testID="avatars-pair-avatars-invitee"
            >
              <Image
                accessibilityComponentType={undefined}
                accessibilityLabel={undefined}
                accessibilityTraits={undefined}
                accessible={true}
                hitSlop={undefined}
                nativeID={undefined}
                onError={undefined}
                onLayout={undefined}
                onLoad={undefined}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                source={
                  Object {
                    "uri": "/var/application/DocumentDir/user-avatar.jpeg",
                  }
                }
                style={
                  Array [
                    Object {
                      "height": 76,
                      "width": 76,
                    },
                    Object {
                      "borderRadius": 38,
                    },
                  ]
                }
                testID={undefined}
              />
            </View>
          </View>
          <Image
            accessibilityLabel="claim-request-check-mark"
            accessible={true}
            source={1}
            style={
              Array [
                Object {
                  "height": 22,
                  "width": 30,
                },
                Object {
                  "height": 20,
                  "right": 7,
                  "width": 80,
                  "zIndex": -1,
                },
              ]
            }
            testID="claim-request-check-mark"
          />
          <View
            accessibilityLabel="claim-request-text-avatars-inviter"
            accessible={true}
            imageShadow={null}
            style={
              Object {
                "borderRadius": 38,
              }
            }
            testID="claim-request-text-avatars-inviter"
          >
            <Image
              accessibilityComponentType={undefined}
              accessibilityLabel={undefined}
              accessibilityTraits={undefined}
              accessible={true}
              hitSlop={undefined}
              nativeID={undefined}
              onError={undefined}
              onLayout={undefined}
              onLoad={undefined}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              source={
                Object {
                  "uri": "http://testissuer.com/logoUrl.png",
                }
              }
              style={
                Array [
                  Object {
                    "height": 76,
                    "width": 76,
                  },
                  Object {
                    "borderRadius": 38,
                  },
                ]
              }
              testID={undefined}
            />
          </View>
        </View>
        <Text
          accessibilityLabel="claim-request-message"
          accessible={true}
          allowFontScaling={false}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "fontSize": 17,
              },
              Object {
                "fontFamily": "Lato",
              },
              Object {
                "backgroundColor": "#ffffff",
                "color": "#777777",
              },
              null,
              Object {
                "textAlign": "center",
              },
              Object {
                "backgroundColor": "transparent",
              },
              null,
              null,
              null,
              null,
              null,
              null,
              null,
              null,
              Object {
                "marginBottom": 5,
              },
            ]
          }
          testID="claim-request-message"
        >
          As soon as
        </Text>
        <Text
          accessibilityLabel="claim-request-message"
          accessible={true}
          allowFontScaling={false}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "fontSize": 17,
              },
              Object {
                "fontFamily": "Lato",
              },
              Object {
                "backgroundColor": "#ffffff",
                "color": "#777777",
              },
              Object {
                "fontWeight": "bold",
              },
              Object {
                "textAlign": "center",
              },
              Object {
                "backgroundColor": "transparent",
              },
              null,
              null,
              null,
              null,
              null,
              null,
              null,
              null,
              Object {
                "marginBottom": 5,
              },
            ]
          }
          testID="claim-request-message"
        />
        <Text
          accessibilityLabel="claim-request-message"
          accessible={true}
          allowFontScaling={false}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "fontSize": 17,
              },
              Object {
                "fontFamily": "Lato",
              },
              Object {
                "backgroundColor": "#ffffff",
                "color": "#777777",
              },
              null,
              Object {
                "textAlign": "center",
              },
              Object {
                "backgroundColor": "transparent",
              },
              null,
              null,
              null,
              null,
              null,
              null,
              null,
              null,
              Object {
                "marginBottom": 5,
              },
            ]
          }
          testID="claim-request-message"
        >
          signs and issues the credential
        </Text>
        <Text
          accessibilityLabel="claim-request-message"
          accessible={true}
          allowFontScaling={false}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "fontSize": 17,
              },
              Object {
                "fontFamily": "Lato",
              },
              Object {
                "backgroundColor": "#ffffff",
                "color": "#777777",
              },
              Object {
                "fontWeight": "bold",
              },
              Object {
                "textAlign": "center",
              },
              Object {
                "backgroundColor": "transparent",
              },
              null,
              null,
              null,
              null,
              null,
              null,
              null,
              null,
              Object {
                "marginBottom": 5,
              },
            ]
          }
          testID="claim-request-message"
        >
          "requester"
        </Text>
        <Text
          accessibilityLabel="claim-request-message"
          accessible={true}
          allowFontScaling={false}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "fontSize": 17,
              },
              Object {
                "fontFamily": "Lato",
              },
              Object {
                "backgroundColor": "#ffffff",
                "color": "#777777",
              },
              null,
              Object {
                "textAlign": "center",
              },
              Object {
                "backgroundColor": "transparent",
              },
              null,
              null,
              null,
              null,
              null,
              null,
              null,
              null,
              Object {
                "marginBottom": 5,
              },
            ]
          }
          testID="claim-request-message"
        >
          to you, it will appear here.
        </Text>
      </View>
      <View
        style={
          Array [
            Object {
              "marginLeft": 15,
              "marginRight": 15,
            },
            undefined,
            Object {
              "marginLeft": 0,
              "marginRight": 0,
            },
            undefined,
          ]
        }
      >
        <View
          accessibilityComponentType={undefined}
          accessibilityLabel={undefined}
          accessibilityTraits={undefined}
          accessible={true}
          hasTVPreferredFocus={undefined}
          hitSlop={undefined}
          isTVSelectable={true}
          nativeID={undefined}
          onLayout={undefined}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Array [
              Object {
                "backgroundColor": "transparent",
              },
              undefined,
            ]
          }
          testID={undefined}
          tvParallaxProperties={undefined}
        >
          <View
            pointerEvents="box-only"
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#9E9E9E",
                  "flexDirection": "row",
                  "justifyContent": "center",
                  "padding": 19,
                },
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
                Object {
                  "backgroundColor": "#ffffff",
                },
                undefined,
                Object {
                  "padding": 12,
                },
                undefined,
                undefined,
                undefined,
                Array [
                  null,
                ],
                undefined,
                undefined,
              ]
            }
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Array [
                  Object {},
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                  Array [
                    Object {
                      "color": "white",
                      "fontSize": 16,
                    },
                    Object {
                      "color": "#4A8FE3",
                    },
                    Object {
                      "fontSize": 14,
                    },
                    undefined,
                    Object {
                      "color": "#85BF43",
                      "fontWeight": "bold",
                    },
                    undefined,
                    undefined,
                    undefined,
                  ],
                ]
              }
            >
              Continue
            </Text>
          </View>
        </View>
      </View>
    </View>
  </View>
</View>
`;
